test_client/tests/src/com/owncloud/android/lib/test_project/test/UpdatePublicShareTest.java

```
@@ -134,7 +139,31 @@ protected void setUp() throws Exception {
 	    	mShare = null;
 	    }
 	    
-		Log.v(LOG_TAG, "Remote fixture created.");
+	 // Create the folder
+	 		mFullPath2FolderToShare = mBaseFolderPath + FOLDER_TO_SHARE;
+	 		result = getActivity().createFolder(
+	 				mFullPath2FolderToShare,
+	 				true);
+	 		if (!result.isSuccess()) {
+	 			Utils.logAndThrow(LOG_TAG, result);
+	 		}
+
+	 		// Share the folder privately with a group
+	 		result = getActivity().createShare(
+	 				mFullPath2FolderToShare,
+	 				ShareType.PUBLIC_LINK,
+	 				"",
+	 				false,
+	 				"",
+	 				OCShare.READ_PERMISSION_FLAG);
+
+	 		if (result.isSuccess()){
+	 			mFolderShare = (OCShare) result.getData().get(0);
+	 		} else{
+	 			mFolderShare = null;
```

In case of fail, should throw an exception here  with ```Utils.logAndThrow```